src/Agent.Listener/JobDispatcher.cs

```diff
@@ -382,10 +408,18 @@ public bool Cancel(JobCancelMessage jobCancelMessage)
                         var completedTask = await Task.WhenAny(renewJobRequest, workerProcessTask, Task.Delay(-1, jobRequestCancellationToken));
                         if (completedTask == workerProcessTask)
                         {
-                            // worker finished successfully, complete job request with result, stop renew lock, job has finished.
+                            // worker finished successfully, complete job request with result, attach unhandled exception reported by worker, stop renew lock, job has finished.
```

Comment:
think about measures here to prevent agent from crashing so that we can still report the error... i dont know